crates/driver/src/domain/competition/solution/slippage.rs

```diff
@@ -0,0 +1,269 @@
+use {
+    crate::domain::{
+        competition::auction::Prices,
+        eth,
+        liquidity::{ExactOutput, MaxInput},
+    },
+    ethcontract::U256,
+    num::{BigRational, CheckedDiv, CheckedMul},
+    number::conversions::big_rational_to_u256,
+    shared::conversions::U256Ext,
+    std::cmp,
+};
+
+#[derive(Clone)]
+pub struct Parameters {
+    /// The maximum relative slippage factor.
+    pub relative: BigRational,
+    /// The maximum absolute slippage in native tokens.
+    pub max: Option<eth::U256>,
+    /// The minimum absolute slippage in native tokens.
+    pub min: Option<eth::U256>,
+    pub prices: Prices,
+}
+
+#[derive(Debug)]
+pub struct Interaction {
+    pub input: eth::Asset,
+    pub output: eth::Asset,
+}
+
+impl Parameters {
+    /// Apply the slippage parameters to the given liquidity increasing the in
+    /// amount by the appropriate slippage while keeping the out amount the
+    /// same.
+    pub fn apply_to(
```

Comment:
Old logic is here:

https://github.com/cowprotocol/services/blob/ab4e96efb6d8642d47e66ad54fc02c41688365d8/crates/solver/src/liquidity/slippage.rs#L44-L89